x-pack/apm-server/main.go

```diff
@@ -15,59 +15,85 @@ import (
 
 	"github.com/elastic/apm-server/beater"
 	"github.com/elastic/apm-server/publish"
+	"github.com/elastic/apm-server/transform"
 	"github.com/elastic/apm-server/x-pack/apm-server/aggregation/txmetrics"
 	"github.com/elastic/apm-server/x-pack/apm-server/cmd"
 )
 
-// runServerWithAggregator runs the APM Server. If aggregation
-// is enabled, then a txmetrics.Aggregator will also be run,
-// and the publish.Reporter will be wrapped such that all
-// transactions pass through the aggregator before being
-// published to libbeat.
-func runServerWithAggregator(ctx context.Context, runServer beater.RunServerFunc, args beater.ServerParams) error {
-	if !args.Config.Aggregation.Enabled {
-		return runServer(ctx, args)
+type namedProcessor struct {
+	name string
+	processor
+}
+
+type processor interface {
+	ProcessTransformables([]transform.Transformable) []transform.Transformable
```

Comment:
why it is a requirement to return a list of transformables? forcing this means that processors can't run in parallel, even when they are independent between each other and from main ingestion